src/double.ts

```diff
@@ -71,13 +72,8 @@ export class Double extends BSONValue {
     return options && options.relaxed ? doubleValue : new Double(doubleValue);
   }
 
-  /** @internal */
-  [Symbol.for('nodejs.util.inspect.custom')](): string {
-    return this.inspect();
-  }
-
-  inspect(): string {
-    const eJSON = this.toExtendedJSON() as DoubleExtended;
-    return `new Double(${eJSON.$numberDouble})`;
+  inspect(depth?: number, options?: unknown, inspect?: InspectParameterFn): string {
+    inspect ??= defaultInspect;
+    return `new Double(${inspect(this.valueOf(), options)})`;
```

Comment:
```suggestion
    return `new Double(${inspect(this.value, options)})`;
```

This is what we have in Int32, and I think there's no need for the indirection of invoking the valueOf() function.

Reply:
Sounds good, I'll change this for my next commit